### PR TITLE
feature: add memory cgroup check when validate config

### PR DESCRIPTION
**Pull request by Ace-Tang**
add two memory cgroup check when create container,
1. --memory-swap should set with --memory
2. value from --memory-swap should greater than --memory.

Signed-off-by: Ace-Tang <aceapril@126.com>

### CODE REVIEW

**Comment by Ace-Tang** [comment]
Without this patch, we can see cgroup not allow the following:

not allow only write memory.memsw.limit
```
#./pouch run -d   --memory-swap=1g 8ac48589692a top
Error: failed to run container 8ab5bb: {"message":"failed to create container(8ab5bbdec429da860444aaa4e9870f462f4c1dec291fa504214a7a3dbec3f9a9) on containerd: failed to create task for container(8ab5bbdec429da860444aaa4e9870f462f4c1dec291fa504214a7a3dbec3f9a9): OCI runtime create failed: container_linux.go:265: starting container process caused \"process_linux.go:348: container init caused \\\"process_linux.go:320: setting cgroup config for procHooks process caused \\\\\\\"failed to write 1073741824 to memory.memsw.limit_in_bytes: write /sys/fs/cgroup/memory/default/8ab5bbdec429da860444aaa4e9870f462f4c1dec291fa504214a7a3dbec3f9a9/memory.memsw.limit_in_bytes: invalid argument\\\\\\\"\\\"\": unknown"}
```

not allow memory.memsw.limit_in_bytes less than memory.limit_in_bytes
```
#./pouch run -d -m=2g  --memory-swap=1g 8ac48589692a top
WARNING: You should typically size your swap space to approximately 2x main memory for systems with less than 2GB of RAM 
Error: failed to run container c2e0a9: {"message":"failed to create container(c2e0a97b9f070f05fa89199a9437ebc102102c83d69f1be892d857ba0809e9fb) on containerd: failed to create task for container(c2e0a97b9f070f05fa89199a9437ebc102102c83d69f1be892d857ba0809e9fb): OCI runtime create failed: container_linux.go:265: starting container process caused \"process_linux.go:348: container init caused \\\"process_linux.go:320: setting cgroup config for procHooks process caused \\\\\\\"failed to write 1073741824 to memory.memsw.limit_in_bytes: write /sys/fs/cgroup/memory/default/c2e0a97b9f070f05fa89199a9437ebc102102c83d69f1be892d857ba0809e9fb/memory.memsw.limit_in_bytes: invalid argument\\\\\\\"\\\"\": unknown"}
```

File: daemon/mgr/container_utils.go [hunk 1]
```diff
@@ -268,6 +268,13 @@ func validateResource(r *types.Resources, update bool) ([]string, error) {
 			warnings = append(warnings, MemorySwapWarn)
 			r.MemorySwap = 0
 		}
+		if r.Memory > 0 && r.MemorySwap > 0 && r.MemorySwap < r.Memory {
+			return warnings, fmt.Errorf("Minimum memoryswap limit should be larger than memory limit")
+		}
+		// cgroup not allow memory-swap less than memory limit
```
Comment (fuweid) [inline daemon/mgr/container_utils.go]: The comment is for the `if r.Memory > 0 && r.MemorySwap > 0 && r.MemorySwap < r.Memory {`?

**Comment by Ace-Tang** [comment]
PTAL, @Letty5411 @fuweid

**Comment by Ace-Tang** [comment]
How about the pr, I think this pr is need to prevent some memory flag uncorrent, @fuweid

**Pull request merged**